tests/gen_cert_test.py

```diff
@@ -181,3 +216,70 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
```

Comment:
We need to pr the code first.  Then we can decide to create a new file.  

tldr; This is not the place for this.

These are the types of comments I'm talking about as being time wasters.  In order to properly answer each of your many points I would also need to write multiple paragraphs.  But this is not a place to have a debate about the state of testing across the platform and how to solve those prevalent issues.  Maybe we should spin up an instance of Reddit and bang our heads against the wall in there.  We can post memes, reference wikipedia articles to prove points, and write endless lists of paraphrased quotes from co-workers when we disagree.

I followed testing paradigms previously done in this repo. As I've seen on many of the PRs I've worked on that have 1000s of lines of test code.  Some of those repos were built recently and can't be blamed on anyone but us.  I balk at this change not because I disagree but because it's coming out of the blue.  And it doesn't make sense to derail this feature even by a day to prioritize fixing a small fraction of the tests in here.  We should put it on a card, prioritize it, and fix the whole file. 

I'm not interested in convincing anyone or arguing my side.  We should be able to have a conversation about the many sides of an issue.  One sided issues are just wrong.